crypto/fipsmodule/aes/mode_wrappers.c

```diff
@@ -148,8 +148,20 @@ int aes_hw_xts_cipher(const uint8_t *in, uint8_t *out, size_t length,
   if (length < 16) return 0;
 
   if (enc) {
+#if defined(AES_XTS_X86_64_AVX512)
+    if (avx512_xts_available()) {
```

Comment:
Thank you for clarifying the comment on the input length.
Do we need to set a minimum input length in this check?
Or the algorithm can handle a single block up to 7 blocks as well as I read in https://github.com/aws/aws-lc/blob/41ab38f71cb63bac051433d04a96f5b869252cb1/crypto/fipsmodule/aes/asm/aesni-xts-avx512.pl#L259?
If the latter, then I suggest adding that to the comment you just edited to clarify the minimum length requirements.
Thank you.

Reply:
we don't need to add a minimum input length check, 1-7 blocks would be handled by do_n_blocks.